### PR TITLE
Engine - yaml configurations for detailed unit testing

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue|
|---|
|#23509|

### Summary
This Pull Request implements Stage 3 of our testing framework development, enhancing the YAML configurations to support detailed unit testing for Wazuh-Engine helper functions. The primary goal is to extend the YAML schema to allow precise specification of inputs and expected outputs, facilitating comprehensive end-to-end testing.

### Changes
- Enhanced the YAML schema to include fields for detailed test specifications such as inputs, expected outputs, and additional testing parameters.
- Updated the test generation tool to parse and utilize the enhanced YAML configurations.
- Developed example YAML configurations to demonstrate new capabilities.
- Validated the updated toolchain with a series of tests to ensure correct interpretation and execution of specified test cases.
- Resolved a bug in the parser combinator of the helper function, ensuring accurate parsing of the extended YAML configurations.

#### YAML Schema Enhancement:

- Defined new fields within the YAML schema for specifying detailed test inputs and expected outputs.
- Added support for complex test cases, including edge cases, error conditions, and varied input scenarios.

#### Test Generation Tool Update:

- Modified the tool to recognize and process the new YAML schema fields.
- Ensured backward compatibility with existing test cases and configurations.

#### Bug Fix:

- Identified and resolved a bug in the parser combinator of the helper function that affected the accurate parsing of extended YAML configurations.
- Implemented additional unit tests to ensure the robustness of the parser combinator following the bug fix.

### CODE REVIEW

File: src/engine/source/builder/src/builders/helperParser.hpp [hunk 1]
```diff
@@ -358,7 +358,8 @@ inline parsec::Parser<HelperToken> getHelperParser(bool eof = false)
     };
 
     auto nameParser = helperNameParser << parenthOpenParser;
-    auto argParserMiddle = getHelperArgParser(argSeparatorParser);
+    auto middleLoolSep = parsec::positiveLook(argSeparatorParser | parenthCloseParser);
```
Comment (JavierBejMen) [inline src/engine/source/builder/src/builders/helperParser.hpp:361]: Can we use a more descriptive name for the Parser? 

Otherwise nice solution.

**Review by JavierBejMen — CHANGES_REQUESTED** [review]

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JavierBejMen — APPROVED** [review]
LGTM!
